### DISCUSSION TITLE
`Launcher` API with `@SpringBootTest`

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3555] [Q&A] [answered]
I'm using `@SpringBootTest` to take advantage of Spring context, and I have a simple structure with a base class with the necessary beans and subclasses with tests.

```java
@SpringBootTest(classes = {...})
public abstract class BaseTest {
  // ...
}
```

```java
public class AuthorizationTest extends BaseTest {

  @Test
   // ...
}
```

Is there any way, using the Launcher API (or other features), to run tests by their method names obtained at runtime before test execution, while being able to use Spring context?

**Comment by sbrannen** [comment, marked as answer]
> Is there any way, using the Launcher API (or other features), to run tests by their method names obtained at runtime before test execution, while being able to use Spring context?

Yes, just use the standard features of the `Launcher` API to select the methods you wish to run.

Or if you're running your tests in an IDE, you can typically right-click on the method you wish to run and select the appropriate menu item to run the test.

If Spring Boot's testing support is configured for the test class, it's also configured for the execution of individual test methods.
